sympy/physics/units/dimensions.py

```diff
@@ -572,3 +572,64 @@ def is_consistent(self):
         # dimensions
         # in vector language: the set of vectors do not form a basis
         return self.inv_can_transf_matrix.is_square
+    def get_dimensionless_numbers(self, list_of_derived_quantities):
+        # list_of_derived_quantities is the input list of derived quantities in the
+        # form of a python dictionary of tuples (key, item) where 'key' is
+        # the name of the physical quantity and 'item' is an instance of
+        # Dimension class which possesses the dimensions of the quantity.
+        """
+        This function is designed to take in a set of physical variables (all 
+    	essentially dimensionful) and returns one set of all possible dimensionless 
+	    quantities, as guided by the Buckingham's pi theorem (a generalization of 
+	    Rayleigh's method of dimensional analysis .....)
+        Please refer and run to the unit testing file '/units/test/test_dimensionLess.py' for
+        the format of the input .... As a heads up, we first need to define a global set of 
+        base dimensions and with those we can define the derived quantities .....
+        """
+        exponent_matrix = []
+        index_dict = {
```

Comment:
This dictionary is not necessary, we already have `self.base_dims`. It's a list, but you could use `self.base_dims.index(dimension)` to make it work the same way as `index_dict`.

Reply:
Thanks for this, but does self.base_dims have a property/attribute by the name index? I looked up the complete file but could not find any such attribute so thought to ask you before pushing in a new commit.

Reply:
Hi @Upabjojr Francesco, I tried removing the redundant whitespaces from the end of each line, and replaced the index_dict with the self.base_dims, and also edited the release notes. But still it shows the two PR tests not working so .....

Reply:
By looking at the logs:

> AssertionError: File contains trailing whitespace: /home/runner/work/sympy/sympy/sympy/physics/units/dimensions.py, line 580.

So, line 580 of file `dimensions.py`.

You can run the code quality tests on your computer by calling `sympy.test("sympy/testing")`

Reply:
Thanks @Upabjojr Francesco, everything else seems to work, the only two errors that remain are the release notes error and the following on the authors' inclusion:

 * Run bin/mailmap_check.py
  bin/mailmap_check.py
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.9.1[2](https://github.com/sympy/sympy/runs/5983251488?check_suite_focus=true#step:6:2)/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/[3](https://github.com/sympy/sympy/runs/5983251488?check_suite_focus=true#step:6:3).9.12/x6[4](https://github.com/sympy/sympy/runs/5983251488?check_suite_focus=true#step:6:4)/lib
This author is not included in the .mailmap file:
Shrohan Mohapatra <4[6](https://github.com/sympy/sympy/runs/5983251488?check_suite_focus=true#step:6:6)8182[7](https://github.com/sympy/sympy/runs/5983251488?check_suite_focus=true#step:6:7)9+ShrohanMohapatra@users.noreply.github.com>

It'd be great if you could suggest something .... Thanks a lot once again :)

Reply:
I wouldn't be too concerned with the mailmap file, once a new SymPy version is released, the mailmap file will be automatically adjusted by taking the author and email list from the git commits. So just make sure your email is correctly inserted into your git commits.

Reply:
Thanks @Upabjojr. I think the write access stays with the mentors, so if everything else works (I think yes my email address is there and correct in the commits), may I be able to request you to merge the 22 commits in the main sympy master branch? I would like to continue to contribute to sympy this summer so ....

Reply:
If the tests don't pass, there is no way to merge this. Merging is only possible after all tests have passed.

There are still some issues with the code that I see right now. I will write later.

Reply:
Okay sure! Awaiting your response ....